libgit2.lua

```diff
@@ -0,0 +1,477 @@
+project "libgit2"
+
+  local prj = project()
+  local prjDir = prj.basedir
+
+  -- -------------------------------------------------------------
+  -- project
+  -- -------------------------------------------------------------
+
+  -- common project settings
+
+  dofile (_BUILD_DIR .. "/3rdparty_static_project.lua")
+
+  -- project specific settings
+
+  uuid "59734853-1767-435a-9969-849cd0d18240"
+
+  flags {
+    "NoPCH",
+  }
+
+  includedirs {
+    prjDir .. "/include/",
+    prjDir .. "/src/",
+    prjDir .. "/deps/http-parser/",
+    prjDir .. "/deps/regex/",
+    prjDir .. "/deps/zlib/"
```

Comment:
Should we use our zlib?

Reply:
Yes. Could you help me understand how? I don't know how to cross-reference 3rd party dependencies with our build system.

Reply:
Sure thing.  You just need to include the zlib dir as seen at https://devtopia.esri.com/3rdparty/libtiff/blob/runtimecore/libtiff.lua#L67.  Half the fun of our 3rdparty libs is that a lot of them use each other.  I try to keep the headers standalone so all needed defines are contained within the library so by adding the zlib include dir and removing the internal one, it should just work.